src/test/regress/specs/isolation_insert_vs_all.spec

```diff
@@ -0,0 +1,83 @@
+# create range distributed table to test behavior of INSERT in concurrent operations
+setup
+{
+	SET citus.shard_replication_factor TO 1;
+    CREATE TABLE insert_hash(id integer, data text);
+	SELECT create_distributed_table('insert_hash', 'id');
+	COPY insert_hash FROM PROGRAM 'echo 0, a\\n1, b\\n2, c\\n3, d\\n4, e' WITH CSV;
+}
+
+# drop distributed table
+teardown
+{
+    DROP TABLE IF EXISTS insert_hash CASCADE;
+}
+
+# session 1
+session "s1"
+step "s1-begin" { BEGIN; }
+step "s1-insert" { INSERT INTO insert_hash VALUES(7, 'k'); }
```

Comment:
They do not block each other even while processing on same shard. I think it is OK to leave it this way.